### DISCUSSION TITLE
Check if window is open

### DISCUSSION THREAD

**Discussion by vismodo** [pokepetter/ursina#288] [Q&A]
How do I check if a window is open? I have a thread checking for the player's position and I want it to stop when the window is closed. I found 2 ways to do this:

- Check if the window is open or not.
- Terminate program when the window is closed.

**Comment by leer-ursina** [comment]
hello,
a window is an object so i will check it like another object
is this check working?=> if win_reference:
it check if window is True

i readed that there is no good and special way to close a thread, they just let the os deal with it
(that is not means there is not...i just found thoses answers)

you could try to transform the thread to a None object before closing:
thread_reference=None
or just close the window
there is method to check how many threads are running and other infos (do not remember it for now sorry)

**Reply by vismodo** [reply]
I don't think you understood my question. I want to check if the window is opened/closed in my thread. All I need to know is how to check if the window is opened or closed.

**Reply by leer-ursina** [reply]
...is "the window" a 3D object??? not a Ursina window??? that is it?!

**Reply by vismodo** [reply]
It is ursina.window

**Reply by vismodo** [reply]
https://www.ursinaengine.org/cheat_sheet.html#window

**Comment by leer-ursina** [comment]
`from ursina import *

win=Ursina()
if win : print("win is open")# your code here
win.run()`

let me know

**Reply by vismodo** [reply]
Nah. Doesn't work.

**Reply by vismodo** [reply]
I mean, It works when the window is open, and when it isn't

**Reply by leer-ursina** [reply]
some code is needed here

here is a way to close thread
[https://www.kite.com/python/answers/how-to-end-a-thread-in-python](url)

**Reply by vismodo** [reply]
yea

**Reply by vismodo** [reply]
I know how to end a thread, I need to know when the window closes to end the thread.
